## Answering Business Questions using SQL

### Introduction and Schema Diagram

The Chinook music store has been purchasing the whole album from the record company but they noticed that there aren't many customers who buy the whole album rather there are more customers who partially buy some songs from the album. The business director asked the Data Analyst team to quantitavely measure the number of cases and create a report in order to make a business decison. 

Here is the Chinook music store's database schema diagram

![tables](table_relations.png)

In [2]:
%%capture
%load_ext sql
%sql sqlite:///chinook.db.sqlite

In [3]:
%%sql
SELECT
    name,
    type
FROM sqlite_master
WHERE type IN ("table","view");

 * sqlite:///chinook.db.sqlite
Done.


[('Album', 'table'),
 ('Artist', 'table'),
 ('Customer', 'table'),
 ('Employee', 'table'),
 ('Genre', 'table'),
 ('Invoice', 'table'),
 ('InvoiceLine', 'table'),
 ('MediaType', 'table'),
 ('Playlist', 'table'),
 ('PlaylistTrack', 'table'),
 ('Track', 'table'),
 ('raw', 'view')]

As you can see above, unlike the shcema diagram, the actual table name and column name are written in pascal case not in snake case.

### Albums vs Individual Tracks

Let's write a query that categorizes each invoice as either an album purchase or not. First of all, let's create a table that shows total number of tracks of each album regardless of customer invoice under the table name of 'num_of_tracks'.

In [56]:
%%sql
CREATE VIEW num_of_track AS
SELECT 
    t.AlbumId AlbumId,
    COUNT(t.TrackId) total_track_num
FROM Track t
GROUP BY 1
ORDER BY 1;

 * sqlite:///chinook.db.sqlite
Done.


[]

In [68]:
%%sql
SELECT * FROM num_of_track;

 * sqlite:///chinook.db.sqlite
Done.


[(1, 10),
 (2, 1),
 (3, 3),
 (4, 8),
 (5, 15),
 (6, 13),
 (7, 12),
 (8, 14),
 (9, 8),
 (10, 14),
 (11, 12),
 (12, 12),
 (13, 8),
 (14, 13),
 (15, 5),
 (16, 7),
 (17, 10),
 (18, 17),
 (19, 11),
 (20, 11),
 (21, 18),
 (22, 3),
 (23, 34),
 (24, 23),
 (25, 13),
 (26, 17),
 (27, 14),
 (28, 10),
 (29, 14),
 (30, 14),
 (31, 9),
 (32, 14),
 (33, 17),
 (34, 17),
 (35, 11),
 (36, 17),
 (37, 20),
 (38, 12),
 (39, 21),
 (40, 12),
 (41, 14),
 (42, 14),
 (43, 7),
 (44, 6),
 (45, 14),
 (46, 13),
 (47, 14),
 (48, 13),
 (49, 10),
 (50, 4),
 (51, 22),
 (52, 15),
 (53, 14),
 (54, 20),
 (55, 20),
 (56, 15),
 (57, 15),
 (58, 9),
 (59, 7),
 (60, 7),
 (61, 11),
 (62, 7),
 (63, 12),
 (64, 9),
 (65, 9),
 (66, 10),
 (67, 16),
 (68, 9),
 (69, 13),
 (70, 13),
 (71, 14),
 (72, 18),
 (73, 30),
 (74, 12),
 (75, 14),
 (76, 15),
 (77, 11),
 (78, 14),
 (79, 10),
 (80, 10),
 (81, 11),
 (82, 13),
 (83, 24),
 (84, 16),
 (85, 14),
 (86, 15),
 (87, 3),
 (88, 12),
 (89, 13),
 (90, 12),
 (91, 16),
 (92, 14),
 (93, 13),
 (94, 11),
 (95, 12),
 (96, 11),
 (97, 10),
 (98, 11),
 (99, 12),
 (100, 9),
 (101, 10),
 (102, 18),
 (103, 10),
 (104, 10),
 (105, 10),
 (106, 9),
 (107, 8),
 (108, 10),
 (109, 9),
 (110, 8),
 (111, 8),
 (112, 8),
 (113, 11),
 (114, 8),
 (115, 20),
 (116, 10),
 (117, 11),
 (118, 11),
 (119, 13),
 (120, 17),
 (121, 10),
 (122, 14),
 (123, 12),
 (124, 14),
 (125, 16),
 (126, 15),
 (127, 10),
 (128, 8),
 (129, 8),
 (130, 7),
 (131, 8),
 (132, 9),
 (133, 9),
 (134, 10),
 (135, 9),
 (136, 7),
 (137, 5),
 (138, 4),
 (139, 15),
 (140, 16),
 (141, 57),
 (142, 14),
 (143, 14),
 (144, 10),
 (145, 18),
 (146, 18),
 (147, 10),
 (148, 12),
 (149, 16),
 (150, 10),
 (151, 14),
 (152, 8),
 (153, 13),
 (154, 8),
 (155, 11),
 (156, 9),
 (157, 14),
 (158, 13),
 (159, 13),
 (160, 15),
 (161, 12),
 (162, 17),
 (163, 17),
 (164, 12),
 (165, 15),
 (166, 14),
 (167, 21),
 (168, 12),
 (169, 16),
 (170, 1),
 (171, 2),
 (172, 1),
 (173, 2),
 (174, 14),
 (175, 12),
 (176, 14),
 (177, 10),
 (178, 16),
 (179, 13),
 (180, 15),
 (181, 11),
 (182, 12),
 (183, 9),
 (184, 16),
 (185, 17),
 (186, 11),
 (187, 11),
 (188, 11),
 (189, 14),
 (190, 16),
 (191, 10),
 (192, 14),
 (193, 17),
 (194, 16),
 (195, 15),
 (196, 14),
 (197, 8),
 (198, 6),
 (199, 12),
 (200, 11),
 (201, 16),
 (202, 18),
 (203, 17),
 (204, 9),
 (205, 10),
 (206, 12),
 (207, 11),
 (208, 7),
 (209, 10),
 (210, 9),
 (211, 18),
 (212, 12),
 (213, 18),
 (214, 11),
 (215, 14),
 (216, 12),
 (217, 14),
 (218, 15),
 (219, 15),
 (220, 11),
 (221, 20),
 (222, 15),
 (223, 15),
 (224, 22),
 (225, 16),
 (226, 1),
 (227, 19),
 (228, 23),
 (229, 26),
 (230, 25),
 (231, 24),
 (232, 12),
 (233, 11),
 (234, 15),
 (235, 11),
 (236, 12),
 (237, 17),
 (238, 14),
 (239, 10),
 (240, 10),
 (241, 14),
 (242, 12),
 (243, 17),
 (244, 11),
 (245, 12),
 (246, 13),
 (247, 15),
 (248, 19),
 (249, 6),
 (250, 22),
 (251, 25),
 (252, 1),
 (253, 24),
 (254, 1),
 (255, 23),
 (256, 12),
 (257, 12),
 (258, 19),
 (259, 17),
 (260, 1),
 (261, 17),
 (262, 2),
 (263, 2),
 (264, 2),
 (265, 2),
 (266, 1),
 (267, 1),
 (268, 1),
 (269, 10),
 (270, 14),
 (271, 14),
 (272, 1),
 (273, 1),
 (274, 1),
 (275, 1),
 (276, 1),
 (277, 1),
 (278, 1),
 (279, 1),
 (280, 2),
 (281, 1),
 (282, 1),
 (283, 1),
 (284, 1),
 (285, 1),
 (286, 1),
 (287, 1),
 (288, 1),
 (289, 1),
 (290, 1),
 (291, 1),
 (292, 1),
 (293, 1),
 (294, 1),
 (295, 1),
 (296, 1),
 (297, 1),
 (298, 1),
 (299, 1),
 (300, 1),
 (301, 1),
 (302, 1),
 (303, 1),
 (304, 1),
 (305, 1),
 (306, 1),
 (307, 1),
 (308, 1),
 (309, 1),
 (310, 1),
 (311, 1),
 (312, 1),
 (313, 1),
 (314, 2),
 (315, 1),
 (316, 1),
 (317, 1),
 (318, 1),
 (319, 1),
 (320, 1),
 (321, 12),
 (322, 11),
 (323, 1),
 (324, 1),
 (325, 1),
 (326, 1),
 (327, 1),
 (328, 1),
 (329, 1),
 (330, 1),
 (331, 1),
 (332, 1),
 (333, 1),
 (334, 1),
 (335, 1),
 (336, 1),
 (337, 1),
 (338, 1),
 (339, 1),
 (340, 1),
 (341, 1),
 (342, 1),
 (343, 1),
 (344, 1),
 (345, 1),
 (346, 1),
 (347, 1)]

Then let's create a table that represents the actual number of tracks per AlbumId in each invoice.

In [54]:
%%sql
CREATE VIEW raw_new AS
SELECT
    i.InvoiceId,
    t.AlbumId,
    COUNT(t.TrackId) actual_track_num
FROM InvoiceLine il
INNER JOIN Invoice i ON i.InvoiceId=il.InvoiceId
INNER JOIN Track t ON t.TrackId=il.TrackId
GROUP BY 1, 2
ORDER BY 1;

 * sqlite:///chinook.db.sqlite
Done.


[]

In [67]:
%%sql
SELECT * FROM raw_new;

 * sqlite:///chinook.db.sqlite
Done.


[(1, 2, 1),
 (1, 3, 1),
 (2, 1, 4),
 (3, 4, 2),
 (3, 5, 4),
 (4, 6, 2),
 (4, 7, 2),
 (4, 8, 2),
 (4, 9, 2),
 (4, 10, 1),
 (5, 11, 2),
 (5, 12, 1),
 (5, 13, 1),
 (5, 14, 1),
 (5, 15, 1),
 (5, 16, 1),
 (5, 17, 1),
 (5, 18, 2),
 (5, 19, 1),
 (5, 20, 1),
 (5, 21, 2),
 (6, 23, 1),
 (7, 23, 2),
 (8, 23, 2),
 (9, 23, 4),
 (10, 24, 6),
 (11, 25, 2),
 (11, 26, 3),
 (11, 27, 2),
 (11, 28, 2),
 (12, 29, 1),
 (12, 30, 2),
 (12, 31, 1),
 (12, 32, 1),
 (12, 33, 2),
 (12, 34, 2),
 (12, 35, 1),
 (12, 36, 2),
 (12, 37, 2),
 (13, 38, 1),
 (14, 38, 2),
 (15, 38, 1),
 (15, 39, 1),
 (16, 39, 4),
 (17, 39, 3),
 (17, 40, 3),
 (18, 23, 2),
 (18, 41, 2),
 (18, 42, 3),
 (18, 43, 1),
 (18, 44, 1),
 (19, 45, 1),
 (19, 46, 2),
 (19, 47, 1),
 (19, 48, 2),
 (19, 49, 1),
 (19, 51, 3),
 (19, 52, 1),
 (19, 53, 2),
 (19, 54, 1),
 (20, 54, 1),
 (21, 55, 2),
 (22, 55, 2),
 (23, 55, 4),
 (24, 55, 1),
 (24, 56, 4),
 (24, 57, 1),
 (25, 57, 2),
 (25, 58, 1),
 (25, 59, 1),
 (25, 60, 1),
 (25, 61, 2),
 (25, 62, 1),
 (25, 63, 1),
 (26, 63, 1),
 (26, 64, 1),
 (26, 65, 1),
 (26, 66, 1),
 (26, 67, 2),
 (26, 68, 1),
 (26, 69, 1),
 (26, 70, 2),
 (26, 71, 1),
 (26, 72, 2),
 (26, 73, 1),
 (27, 74, 1),
 (28, 74, 2),
 (29, 74, 2),
 (30, 74, 1),
 (30, 75, 3),
 (31, 75, 2),
 (31, 76, 3),
 (31, 77, 1),
 (32, 77, 1),
 (32, 78, 3),
 (32, 79, 1),
 (32, 80, 2),
 (32, 81, 2),
 (33, 73, 2),
 (33, 82, 1),
 (33, 83, 3),
 (33, 84, 2),
 (33, 85, 1),
 (33, 86, 2),
 (33, 88, 1),
 (33, 89, 2),
 (34, 91, 1),
 (35, 91, 2),
 (36, 91, 2),
 (37, 91, 4),
 (38, 92, 3),
 (38, 93, 3),
 (39, 94, 2),
 (39, 95, 2),
 (39, 96, 2),
 (39, 97, 2),
 (39, 98, 1),
 (40, 99, 1),
 (40, 100, 1),
 (40, 101, 2),
 (40, 102, 2),
 (40, 103, 1),
 (40, 104, 1),
 (40, 105, 1),
 (40, 106, 1),
 (40, 107, 1),
 (40, 108, 1),
 (40, 109, 1),
 (40, 110, 1),
 (41, 112, 1),
 (42, 112, 2),
 (43, 112, 1),
 (43, 113, 1),
 (44, 113, 4),
 (45, 114, 2),
 (45, 115, 4),
 (46, 116, 2),
 (46, 117, 2),
 (46, 118, 2),
 (46, 119, 2),
 (46, 120, 1),
 (47, 120, 1),
 (47, 121, 1),
 (47, 122, 2),
 (47, 123, 1),
 (47, 124, 2),
 (47, 125, 1),
 (47, 126, 2),
 (47, 127, 1),
 (47, 128, 1),
 (47, 129, 1),
 (47, 130, 1),
 (48, 132, 1),
 (49, 132, 2),
 (50, 132, 1),
 (50, 133, 1),
 (51, 133, 3),
 (51, 134, 1),
 (52, 134, 2),
 (52, 135, 2),
 (52, 136, 2),
 (53, 137, 1),
 (53, 139, 3),
 (53, 140, 2),
 (53, 141, 3),
 (54, 142, 1),
 (54, 143, 2),
 (54, 144, 1),
 (54, 145, 2),
 (54, 146, 2),
 (54, 147, 1),
 (54, 148, 1),
 (54, 149, 2),
 (54, 150, 1),
 (54, 151, 1),
 (55, 152, 1),
 (56, 152, 2),
 (57, 152, 2),
 (58, 153, 4),
 (59, 153, 1),
 (59, 154, 2),
 (59, 155, 3),
 (60, 156, 1),
 (60, 157, 2),
 (60, 158, 3),
 (60, 159, 2),
 (60, 160, 1),
 (61, 160, 1),
 (61, 161, 1),
 (61, 162, 2),
 (61, 163, 2),
 (61, 164, 1),
 (61, 165, 2),
 (61, 166, 1),
 (61, 167, 3),
 (61, 168, 1),
 (62, 169, 1),
 (63, 169, 2),
 (64, 169, 2),
 (65, 171, 1),
 (65, 172, 1),
 (65, 173, 1),
 (65, 174, 1),
 (66, 174, 3),
 (66, 175, 3),
 (67, 176, 2),
 (67, 177, 2),
 (67, 178, 2),
 (67, 179, 2),
 (67, 180, 1),
 (68, 141, 1),
 (68, 180, 1),
 (68, 181, 1),
 (68, 182, 2),
 (68, 183, 1),
 (68, 184, 2),
 (68, 185, 2),
 (68, 186, 1),
 (68, 188, 2),
 (68, 189, 1),
 (69, 190, 1),
 (70, 190, 2),
 (71, 190, 2),
 (72, 190, 4),
 (73, 191, 2),
 (73, 192, 4),
 (74, 193, 3),
 (74, 194, 2),
 (74, 195, 3),
 (74, 196, 1),
 (75, 141, 2),
 (75, 196, 1),
 (75, 198, 1),
 (75, 199, 1),
 (75, 200, 1),
 (75, 201, 2),
 (75, 202, 2),
 (75, 203, 2),
 (75, 204, 1),
 (75, 205, 1),
 (76, 206, 1),
 (77, 206, 2),
 (78, 207, 2),
 (79, 207, 4),
 (80, 208, 1),
 (80, 209, 3),
 (80, 210, 2),
 (81, 211, 3),
 (81, 212, 2),
 (81, 213, 3),
 (81, 214, 1),
 (82, 215, 2),
 (82, 216, 1),
 (82, 217, 2),
 (82, 218, 1),
 (82, 219, 2),
 (82, 220, 1),
 (82, 221, 3),
 (82, 222, 1),
 (82, 223, 1),
 (83, 224, 1),
 (84, 224, 2),
 (85, 224, 2),
 (86, 224, 4),
 (87, 224, 1),
 (87, 225, 4),
 (87, 227, 1),
 (88, 227, 3),
 (88, 228, 3),
 (88, 229, 2),
 (88, 230, 1),
 (89, 230, 3),
 (89, 231, 2),
 (89, 232, 2),
 (89, 233, 1),
 (89, 234, 1),
 (89, 235, 2

As a side note, for example, invoice number #1 consists of one track of album #2 and one track of album #3. In order to compare whether it is a whole album purchase or not, let's join the 'num_of_track' table into 'raw_new' table.

In [69]:
%%sql
CREATE VIEW new_table AS
SELECT
    r.*,
    nt.total_track_num,
    CASE WHEN actual_track_num=total_track_num THEN '1' ELSE '0' END AS condition
FROM raw_new r
INNER JOIN num_of_track nt ON r.AlbumId=nt.AlbumId
ORDER BY 1;

 * sqlite:///chinook.db.sqlite
Done.


[]

In [70]:
%%sql
SELECT * FROM new_table;

 * sqlite:///chinook.db.sqlite
Done.


[(1, 2, 1, 1, '1'),
 (1, 3, 1, 3, '0'),
 (2, 1, 4, 10, '0'),
 (3, 4, 2, 8, '0'),
 (3, 5, 4, 15, '0'),
 (4, 6, 2, 13, '0'),
 (4, 7, 2, 12, '0'),
 (4, 8, 2, 14, '0'),
 (4, 9, 2, 8, '0'),
 (4, 10, 1, 14, '0'),
 (5, 11, 2, 12, '0'),
 (5, 12, 1, 12, '0'),
 (5, 13, 1, 8, '0'),
 (5, 14, 1, 13, '0'),
 (5, 15, 1, 5, '0'),
 (5, 16, 1, 7, '0'),
 (5, 17, 1, 10, '0'),
 (5, 18, 2, 17, '0'),
 (5, 19, 1, 11, '0'),
 (5, 20, 1, 11, '0'),
 (5, 21, 2, 18, '0'),
 (6, 23, 1, 34, '0'),
 (7, 23, 2, 34, '0'),
 (8, 23, 2, 34, '0'),
 (9, 23, 4, 34, '0'),
 (10, 24, 6, 23, '0'),
 (11, 25, 2, 13, '0'),
 (11, 26, 3, 17, '0'),
 (11, 27, 2, 14, '0'),
 (11, 28, 2, 10, '0'),
 (12, 29, 1, 14, '0'),
 (12, 30, 2, 14, '0'),
 (12, 31, 1, 9, '0'),
 (12, 32, 1, 14, '0'),
 (12, 33, 2, 17, '0'),
 (12, 34, 2, 17, '0'),
 (12, 35, 1, 11, '0'),
 (12, 36, 2, 17, '0'),
 (12, 37, 2, 20, '0'),
 (13, 38, 1, 12, '0'),
 (14, 38, 2, 12, '0'),
 (15, 38, 1, 12, '0'),
 (15, 39, 1, 21, '0'),
 (16, 39, 4, 21, '0'),
 (17, 39, 3, 21, '0'),
 (17, 40, 3, 12, '0'),
 (18, 23, 2, 34, '0'),
 (18, 41, 2, 14, '0'),
 (18, 42, 3, 14, '0'),
 (18, 43, 1, 7, '0'),
 (18, 44, 1, 6, '0'),
 (19, 45, 1, 14, '0'),
 (19, 46, 2, 13, '0'),
 (19, 47, 1, 14, '0'),
 (19, 48, 2, 13, '0'),
 (19, 49, 1, 10, '0'),
 (19, 51, 3, 22, '0'),
 (19, 52, 1, 15, '0'),
 (19, 53, 2, 14, '0'),
 (19, 54, 1, 20, '0'),
 (20, 54, 1, 20, '0'),
 (21, 55, 2, 20, '0'),
 (22, 55, 2, 20, '0'),
 (23, 55, 4, 20, '0'),
 (24, 55, 1, 20, '0'),
 (24, 56, 4, 15, '0'),
 (24, 57, 1, 15, '0'),
 (25, 57, 2, 15, '0'),
 (25, 58, 1, 9, '0'),
 (25, 59, 1, 7, '0'),
 (25, 60, 1, 7, '0'),
 (25, 61, 2, 11, '0'),
 (25, 62, 1, 7, '0'),
 (25, 63, 1, 12, '0'),
 (26, 63, 1, 12, '0'),
 (26, 64, 1, 9, '0'),
 (26, 65, 1, 9, '0'),
 (26, 66, 1, 10, '0'),
 (26, 67, 2, 16, '0'),
 (26, 68, 1, 9, '0'),
 (26, 69, 1, 13, '0'),
 (26, 70, 2, 13, '0'),
 (26, 71, 1, 14, '0'),
 (26, 72, 2, 18, '0'),
 (26, 73, 1, 30, '0'),
 (27, 74, 1, 12, '0'),
 (28, 74, 2, 12, '0'),
 (29, 74, 2, 12, '0'),
 (30, 74, 1, 12, '0'),
 (30, 75, 3, 14, '0'),
 (31, 75, 2, 14, '0'),
 (31, 76, 3, 15, '0'),
 (31, 77, 1, 11, '0'),
 (32, 77, 1, 11, '0'),
 (32, 78, 3, 14, '0'),
 (32, 79, 1, 10, '0'),
 (32, 80, 2, 10, '0'),
 (32, 81, 2, 11, '0'),
 (33, 73, 2, 30, '0'),
 (33, 82, 1, 13, '0'),
 (33, 83, 3, 24, '0'),
 (33, 84, 2, 16, '0'),
 (33, 85, 1, 14, '0'),
 (33, 86, 2, 15, '0'),
 (33, 88, 1, 12, '0'),
 (33, 89, 2, 13, '0'),
 (34, 91, 1, 16, '0'),
 (35, 91, 2, 16, '0'),
 (36, 91, 2, 16, '0'),
 (37, 91, 4, 16, '0'),
 (38, 92, 3, 14, '0'),
 (38, 93, 3, 13, '0'),
 (39, 94, 2, 11, '0'),
 (39, 95, 2, 12, '0'),
 (39, 96, 2, 11, '0'),
 (39, 97, 2, 10, '0'),
 (39, 98, 1, 11, '0'),
 (40, 99, 1, 12, '0'),
 (40, 100, 1, 9, '0'),
 (40, 101, 2, 10, '0'),
 (40, 102, 2, 18, '0'),
 (40, 103, 1, 10, '0'),
 (40, 104, 1, 10, '0'),
 (40, 105, 1, 10, '0'),
 (40, 106, 1, 9, '0'),
 (40, 107, 1, 8, '0'),
 (40, 108, 1, 10, '0'),
 (40, 109, 1, 9, '0'),
 (40, 110, 1, 8, '0'),
 (41, 112, 1, 8, '0'),
 (42, 112, 2, 8, '0'),
 (43, 112, 1, 8, '0'),
 (43, 113, 1, 11, '0'),
 (44, 113, 4, 11, '0'),
 (45, 114, 2, 8, '0'),
 (45, 115, 4, 20, '0'),
 (46, 116, 2, 10, '0'),
 (46, 117, 2, 11, '0'),
 (46, 118, 2, 11, '0'),
 (46, 119, 2, 13, '0'),
 (46, 120, 1, 17, '0'),
 (47, 120, 1, 17, '0'),
 (47, 121, 1, 10, '0'),
 (47, 122, 2, 14, '0'),
 (47, 123, 1, 12, '0'),
 (47, 124, 2, 14, '0'),
 (47, 125, 1, 16, '0'),
 (47, 126, 2, 15, '0'),
 (47, 127, 1, 10, '0'),
 (47, 128, 1, 8, '0'),
 (47, 129, 1, 8, '0'),
 (47, 130, 1, 7, '0'),
 (48, 132, 1, 9, '0'),
 (49, 132, 2, 9, '0'),
 (50, 132, 1, 9, '0'),
 (50, 133, 1, 9, '0'),
 (51, 133, 3, 9, '0'),
 (51, 134, 1, 10, '0'),
 (52, 134, 2, 10, '0'),
 (52, 135, 2, 9, '0'),
 (52, 136, 2, 7, '0'),
 (53, 137, 1, 5, '0'),
 (53, 139, 3, 15, '0'),
 (53, 140, 2, 16, '0'),
 (53, 141, 3, 57, '0'),
 (54, 142, 1, 14, '0'),
 (54, 143, 2, 14, '0'),
 (54, 144, 1, 10, '0'),
 (54, 145, 2, 18, '0'),
 (54, 146, 2, 18, '0'),
 (54, 147, 1, 10, '0'),
 (54, 148, 1, 12, '0'),
 (54, 149, 2, 16, '0'),
 (54, 150, 1, 10, '0'),
 (54, 151, 1,

Now we can figure out which invoice contains full album purchase by referring a 'condition' column ('1': full album purchase, '0': not full album purchase)

In [75]:
%%sql
SELECT 
    SUM(condition) total_whole_ablum_purchase_case,
    COUNT(AlbumId)
FROM new_table;

 * sqlite:///chinook.db.sqlite
Done.


[(49, 1303)]

There are 49 cases of whole album purchase while there are 1303 cases of album purchase (3.3% = 49/1303). Based on this data, I would recommend against purchasing whole tracks from albums from record companies, since most of the customer invoice consist of individual track purchase.